configure-hugepages/configure-hugepages.yaml

```diff
@@ -0,0 +1,116 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Deploy this DaemonSet to configure hugepages on nodes with the
+# "cloud.google.com/gke-configure-hugepages=true" label.
+#
+# Change the values of NR_HUGEPAGES and HUGEPAGE_SIZE to suit your needs. See
+# https://www.kernel.org/doc/Documentation/vm/hugetlbpage.txt for details.
+# This daemonset currently only supports 1G and 2M hugepages. The task will
+# also fail if the node it's run on does not have enough free memory for the
+# kernel to allocate the requested number of hugepages with the specified size.
+#
+# WARNING: Changing the hugepages requires a kubelet restart. Therefore, in
+# order to avoid disrupting your workloads, it is recommended to create a new
+# node pool with the "cloud.google.com/gke-configure-hugepages=true" label in your
+# cluster, deploy the DaemonSet to configure hugepages in that node pool,
+# and then migrate your workloads to the new node pool.
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: gke-configure-hugepages
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      name: gke-configure-hugepages
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        name: gke-configure-hugepages
+    spec:
+      tolerations:
+      - operator: Exists
+      volumes:
+      - name: host
+        hostPath:
+          path: /
+      hostPID: true
+      initContainers:
+      - name: gke-configure-hugepages
+        image: "gke.gcr.io/gke-distroless/bash"
+        env:
+        # Number of 2M hugepages. Update the value as desired.
+        - name: NR_HUGEPAGES_2M
+          value: "100"
```

Comment:
can we set the default value to 0 for both? it's unlikely user need both and we should fail if both are 0 meaning users didn't specify a value.

Reply:
Sure, that makes sense. I only set them to nonzero as an example with the expectation it would be modified, but to your point, 0 is just as good of an example as the arbitrary numbers I chose :smile: